src/main/java/org/opensearch/security/configuration/DlsFlsFilterLeafReader.java

```diff
@@ -126,7 +126,7 @@ class DlsFlsFilterLeafReader extends SequentialStoredFieldsLeafReader {
                 this.flsFieldInfos = delegate.getFieldInfos();
             }
 
-            dge = new DlsGetEvaluator(dlsQuery, in, applyDlsHere());
+            dge = new DlsGetEvaluator(dlsQuery, in, dlsQuery != null && applyDlsHere());
```

Comment:
Nit: Do we need the additional check here? The same check is done right inside, in line 142.

Reply:
it is required because it prevents `applyDlsHere()` from being called if dlsQuery is null. The call to `applyDlsHere()` fails with an assertion error otherwise.